### PR TITLE
Add two way binding to slider

### PR DESCRIPTION
**Pull request by iveysaur**
R: @jelbourn @kara

### CODE REVIEW

File: src/components/slider/slider.ts [hunk 1]
```diff
@@ -6,7 +6,13 @@ import {
   Input,
   ViewEncapsulation,
   AfterContentInit,
+  forwardRef,
 } from '@angular/core';
+import {
+    NG_VALUE_ACCESSOR,
+    ControlValueAccessor,
+    FormsModule,
+} from '@angular/forms';
```
Comment (jelbourn) [inline src/components/slider/slider.ts]: Make import indent consistent in this file

File: src/demo-app/demo-app-module.ts [hunk 2]
```diff
@@ -37,6 +37,7 @@ import {TabsDemo} from './tabs/tab-group-demo';
   imports: [
     BrowserModule,
     FormsModule,
+    ReactiveFormsModule,
```
Comment (kara) [inline src/demo-app/demo-app-module.ts]: Given that in the demo, you ended up with `ngModel`, you can remove `ReactiveFormsModule` from the demo module.

File: src/components/slider/slider.ts [hunk 3]
```diff
@@ -138,7 +167,8 @@ export class MdSlider implements AfterContentInit {
    */
   ngAfterContentInit() {
     this._sliderDimensions = this._renderer.getSliderDimensions();
-    this.snapToValue();
+    this._controlValueAccessorChangeFn(this.value);
```
Comment (jelbourn) [inline src/components/slider/slider.ts]: Add comment explaining why this is necessary

File: src/components/slider/slider.spec.ts [hunk 4]
```diff
@@ -588,6 +590,65 @@ describe('MdSlider', () => {
       expect(sliderContainerElement.classList).toContain('md-slider-active');
     });
   });
+
+  describe('slider with two-way binding', () => {
```
Comment (kara) [inline src/components/slider/slider.spec.ts]: You've got a one-way binding in this test :)   Pet peeve, but can you change to read "slider as a custom form control"?

**Comment by jelbourn** [comment]
LGTM aside from comments aside from mine and Kara's comments.

File: src/components/slider/slider.ts [hunk 5]
```diff
@@ -315,6 +345,34 @@ export class MdSlider implements AfterContentInit {
...
+  registerOnChange(fn: (value: any) => void) {
+    this._controlValueAccessorChangeFn = fn;
+  }
+
+  /**
+   * Implemented as part of ControlValueAccessor.
+   * TODO: internal
+   */
+  registerOnTouched(fn: any) {
+    this.onTouched = fn;
```
Comment (kara) [inline src/components/slider/slider.ts]: Where are you calling this?   It doesn't look like the onBlur function calls it.  Have you checked that the control is correctly set as touched?

**Comment by kara** [comment]
LGTM

**Pull request merged**